model_api/python/model_api/models/model.py

```diff
@@ -402,6 +404,40 @@ def __call__(self, inputs):
         raw_result = self.infer_sync(dict_data)
         return self.postprocess(raw_result, input_meta)
 
+    def infer_batch(self, inputs):
+        """
+        Applies preprocessing, asynchronous inference, postprocessing routines to a collection of inputs.
+
+        Args:
+            inputs (list): a list of inputs for inference
+
+        Returns:
+            list: a list of inference results
+        """
+        self.await_all()
+
+        completed_results = {}
+
+        @contextmanager
+        def tmp_callback(*args, **kwds):
+            old_callback = self.callback_fn
```

Comment:
If user set own callbacks, then it's ignored. Is it ok?

Reply:
This functionality saves user's callback, and then restores it

Reply:
User's callback is not supposed to be executed during batch inference, so we have to backup it for the future